### ISSUE TITLE
derivation parser error on the sets of scopes expander

### ISSUE THREAD

**Issue by florence** [racket/macro-debugger#5]
If you attempt to macro step the below program the stepper errors with an internal error:

```
Internal error:
derivation-parser: error on token #5860: <syntax-error, #(struct:exn:fail:syntax:unbound "id: unbound
identifier in module" #<continuation-mark-set> (#<syntax:/Users/florence/tmp/stx.rkt:25:34 id>))>
```

The program is:

```racket
#lang racket
(require (for-syntax syntax/parse))

;; a standard context for identifiers
(define-for-syntax ctx #'ctx)

;; create an ID with the context `ctx`, and the current
;; expander mark (so that the mark is canceled later),
;; and the location loc
(define-for-syntax (make-id loc)
  (syntax-local-introduce
   (datum->syntax ctx 'id loc)))

;; This introduces a binding
(define-syntax (def stx)
  (syntax-parse stx
    [(def)
     (with-syntax ([id (make-id #'here)])
       #'(define id 5))]))

;; this attempts to use the binding introduced by `def`
(define-syntax (use stx)
  (syntax-parse stx
    [(use)
     (with-syntax ([id (make-id #'here)])
       #'id)]))

;; this works
#;
(begin
  (def)
  (use))

;; this fails
(let ()
  (def)
  (use))
```

**Comment by dfeltey** [comment]
Here's another program that causes this issue:

```
#lang racket

(require (for-syntax syntax/parse))

(define-syntax (bad-let stx)
  (syntax-parse stx
     [(_ body ...)
     #`(let ()
         ;; Reference an unbound identifier
         bad-var
         body ...)]))

(bad-let 5)
```

With the error:

```
Internal error:
derivation-parser: error on token #2629: <syntax-error, #(struct:exn:fail:syntax:unbound "bad-var: unbound identifier in module" #<continuation-mark-set> (#<syntax:10:9 bad-var>))>
```

**Comment by rmculpepper** [comment]
Fixed by cdffe710 and racket/racket@a41c63be.
